File: August-LeetCoding-Challenge/24-Sum-of-Left-Leaves/Sum-of-Left-Leaves.py
class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        if not root:
            return 0
        if not root.left and not root.right:
            return 0
        q = [[root,"mid"]]
        res = 0
        while q != []:
            l = []
            while q:
                l.append(q.pop(0))
            for each in l:
                x = each[0]
                if not x.left and not x.right:
                    if each[1]=="left":
                        res += x.val
                if x.left:
                    q.append([x.left,"left"])
                if x.right:
                    q.append([x.right,"right"])
        return res

File: August-LeetCoding-Challenge/24-Sum-of-Left-Leaves/Sum-of-Left-Leaves.cs
public class Solution {
    public int SumOfLeftLeaves(TreeNode root) {
        if(root == null) return 0;
        
        Dfs(root.left, 0);
        Dfs(root.right, 1);
        return sum;
    }
    private int sum =0;
    private void Dfs(TreeNode node, int direction){
        if(node == null) return;
        
        if(node.left == null && node.right == null && direction == 0){
            sum += node.val;
        }
        
        Dfs(node.left, 0);
        Dfs(node.right, 1);        
    }
}
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */